src/tests/cwrapper/test_cwrapper.c

```diff
@@ -0,0 +1,40 @@
+#include <stdio.h>
+#include "cwrapper.h"
+
+void test_cwrapper() {
+    char* s;
+    basic x = basic_new(), y = basic_new(), z = basic_new();
+    symbol_set(x, "x");
+    symbol_set(y, "y");
+    symbol_set(z, "z");
+    
+    s = basic_str(x);
+    printf("Symbol : %s\n", s);
+    basic e = basic_new();
+
+    integer_set_ui(e, 123);
+    s = basic_str(e);
+    printf("Integer : %s\n", s);
+
+    integer_set_ui(e, 456);
+    basic_add(e, e, x);
+    basic_mul(e, e, y);
+    basic_div(e, e, z);
+    printf("Basic : %s\n", basic_str(e));
+    basic_diff(e, e, z);
+    printf("Basic : %s\n", basic_str(e));
+
+    rational_set_ui(e, 100, 47);
+    s = basic_str(e);
+
+    printf("Rational : %s\n", s);
+    printf("Is_a_Symbol %s: %d\n", s, is_a_Symbol(e));
+    printf("Is_a_Rational %s: %d\n", s, is_a_Rational(e));
+    printf("Is_a_Integer %s: %d\n", s, is_a_Integer(e));
```

Comment:
God it, the final trick is this:
```diff
diff --git a/src/cwrapper.cpp b/src/cwrapper.cpp
index 9a70abf..6918226 100644
--- a/src/cwrapper.cpp
+++ b/src/cwrapper.cpp
@@ -26,6 +26,7 @@ extern "C" {
 void basic_free(basic s)
 {
     s->ptr.reset();
+    delete s;
 }
 
 basic basic_new()
```
Then there are no leaks:
```
certik@redhawk:~/repos/csympy(pr-357)$ valgrind --leak-check=full ./src/tests/cwrapper/test_cwrapper 
==29482== Memcheck, a memory error detector
==29482== Copyright (C) 2002-2012, and GNU GPL'd, by Julian Seward et al.
==29482== Using Valgrind-3.8.1 and LibVEX; rerun with -h for copyright info
==29482== Command: ./src/tests/cwrapper/test_cwrapper
==29482== 
==29482== 
==29482== HEAP SUMMARY:
==29482==     in use at exit: 0 bytes in 0 blocks
==29482==   total heap usage: 2,665 allocs, 2,665 frees, 48,523 bytes allocated
==29482== 
==29482== All heap blocks were freed -- no leaks are possible
==29482== 
==29482== For counts of detected and suppressed errors, rerun with: -v
==29482== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 6 from 6)
```

Would you mind fixing the above?

The string handling should be fixed by deallocating the string pointer.